rfcs/text/pe/pe.yml

```diff
@@ -0,0 +1,193 @@
+---
+- name: pe
+  type: group
+  fields:
+    - name: main_icon
+      level: extended
+      type: object
+      description: >
+        Hashes of embedded program icon.
+      fields:
+        - name: dhash
+          level: extended
+          type: keyword
+          description: >
+            Difference Hash (dhash) for a given PE file.
+          example: b806e17c8e330d82
+
+        - name: md5
+          level: extended
+          type: keyword
+          description: >
+            MD5 hash of raw icon data.
+          example: 6d1cae6272afbb88876ed6476b990d8c
+
+    - name: debug
+      level: extended
+      type: keyword
+      description: Debug information, if present
+      example: { "offset" : 1296336, "size" : 816, "type_str" : "IMAGE_DEBUG_TYPE_POGO", "type" : 13, "timestamp" : "Wed Oct 21 09:01:33 2020" }
+
+    - name: import_list
+      level: extended
+      type: keyword
+      description: List of all imported functions
+      example: { "library_name" : "mscoree.dll", "imported_functions" : "GetFileVersionInfoSizeA" }
+
+    - name: sections
+      level: extended
+      description: >
+        Data about sections of compiled binary PE
+      type: keyword
+      example: { "chi2" : 3027194, "virtual_address": 8192, "entropy": 6.24, "flags": "rx", "name": ".text", "raw_size": 198144 }
+
+    - name: resource_details
+      level: extended
+      type: keyword
+      description: >
+        If the PE contains resources, some info about them
+      example: { "chi2": -1, "filetype": "English text", "entropy": 0, "sha256": "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855", "lang": "CHINESE SIMPLIFIED" }
+
+    - name: resource_languages
+      level: extended
+      type: keyword
+      short: List of resource languages.
+      description: >
+        Digest of languages found in resources. Key is language (as string) and
+        value is how many resources there are having that language (as integer)
+      example: { "ENGLISH US": 1, "CHINESE SIMPLIFIED": 760 }
```

Comment:
Could you go through and update these examples? Pretty sure as this is a `type: keyword` the example is actually:

`[ "ENGLISH US", "CHINESE SIMPLIFIED" ]`

Reply:
In this case, it provides a summary of each of the resource languages and VT does return it as presented here. that said, I think it's not incredibly useful in this format. A list of keywords as suggested by @andrewstucki would give us a single point to search to see if _any_ resource matches a given language set. But then, we could also do that with the resource list and a nested query. In that case, this could be eliminated altogether or be restructured as:

```json
[ {"name": "ENGLISH US", "count": 1 }, {"name": "CHINESE SIMPLIFIED", "count": 760}]```

Without one or the other, you'll have a field name explosion to account for all possible localizations.

Reply:
@dcode I didn't know those were counts, what is it telling us that we'd not to represent it with?
```
[ "ENGLISH US", "CHINESE SIMPLIFIED" ]
```

Reply:
@peasead yeah, so those are the rolled up totals of the language code for each resource, retrieved from traversing the resource table. Each leaf node of the resource table corresponds to a single embedded resource and has parent nodes that specify the language of each of the given resources. It's basically just an aggregate of the individual resources that look something like this on their own:

https://github.com/andrewstucki/fingerprint/blob/625f616f4551c33d5e984ab28a0cc48e793c7fad/fixtures/calc.exe.fingerprint#L508-L512

I was more or less commenting on the description of the field and the type. It's currently a keyword, so I would think this would just be a list of all seen language codes acting as a "digest". That said, I'm not quite sure we actually need this anyway, since the data would exist on all of the individual resources anyway and I'm not sure that ECS is where we'd want to create extra fields just for pre-rolled aggregates.

Reply:
I vote drop this field entirely since it can be equally aggregated with nested agg

Reply:
Done.